packages/lesswrong/server/repos/TweetsRepo.ts

```diff
@@ -0,0 +1,69 @@
+import AbstractRepo from "./AbstractRepo";
+import Tweets from "../../lib/collections/tweets/collection";
+import { recordPerfMetrics } from "./perfMetricWrapper";
+import { getViewablePostsSelector } from "./helpers";
+
+class TweetsRepo extends AbstractRepo<"Tweets"> {
+  constructor() {
+    super(Tweets);
+  }
+
+  async getUntweetedPostsCrossingKarmaThreshold({threshold, since}: {threshold: number, since: Date}): Promise<string[]> {
+    const res = await this.getRawDb().any<{postId: string}>(`
+      -- TweetsRepo.getPostsCrossingKarmaThreshold
+      WITH KarmaChanges AS (
+        SELECT
+          v."votedAt",
+          v."documentId" AS post_id,
+          v.power,
+          (
+            SELECT
+              sum(v1.power) AS karma
+          FROM
+            "Posts" p1
+          LEFT JOIN "Votes" v1 ON p1._id = v1."documentId"
+        WHERE
+          p1._id = v."documentId"
+          AND v1.cancelled IS FALSE
+          AND v1."votedAt" < v."votedAt"
+        GROUP BY
+          p1._id) AS prior_karma,
+        (
+          SELECT
+            sum(v1.power) AS karma
+          FROM
+            "Posts" p1
+          LEFT JOIN "Votes" v1 ON p1._id = v1."documentId"
+        WHERE
+          p1._id = v."documentId"
+          AND v1.cancelled IS FALSE
+          AND v1."votedAt" <= v."votedAt"
+        GROUP BY
+          p1._id) AS post_karma
```

Comment:
We can, I'll change it to that. The reason I didn't initially is because it requires an extra layer of nesting in the query, but it's still probably more readable with the nesting